PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -4,7 +4,13 @@
  #                                                 #
  ##################################################>
 
- Import-Module Storage
+Import-Module Storage
+Import-Module SmbShare
+Import-Module SmbWitness
+Import-Module FailoverClusters
+Import-Module CimCmdlets
+Import-Module StorageSubSystem
+Add-Type -assembly "system.io.compression.filesystem"
```

Comment:
Is this actually necessary? Look further down the script. One or the other should go?

Reply:
Just to be clear, I'm referring to the Add-Type.

Reply:
you can remove the +Add-Type -assembly "system.io.compression.filesystem" if you wish

Reply:
Does it work for you if I do? Thats why I'm asking, I don't know how to re-create the environment this level of dependency specificity is required for.

There are at least three other modules I know this uses, at least one of which seems like it should fall into this category: NetAdapter (Get-Net*) and DcbQos (the Get-NetQosPolicy). Do we have all of them? I'd really like to hear/confirm that this runs successfully as-is despite that, so I know how to make sure we get additional dependencies right the first time going forward.

Reply:
Lets chat on Monday and I can show you in my lab and we can discuss the best way forward.